src/Integration.Vsix/SonarLintDaemon/SonarLintDaemon.cs

```diff
@@ -441,5 +435,42 @@ internal void SafeOperation(Action op)
                 logger.WriteLine($"Daemon error: {ex.ToString()}");
             }
         }
+
+        #region IDisposable Support
+        private bool disposedValue = false; // To detect redundant calls
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
```

Comment:
shouldn't `if (disposing)` be the outer if? because now, if `!disposedValue` and `!disposing`, `disposedValue` will be set to `true` - shouldn't it happen only if `disposing`?